src/MainWindow.cpp

```diff
@@ -1335,7 +1335,7 @@ void MainWindow::dataTableSelectionChanged(const QModelIndex& index)
     }
 
     bool editingAllowed = !db.readOnly() && (m_currentTabTableModel == m_browseTableModel) &&
-            (db.getObjectByName(currentlyBrowsedTableName())->type() == sqlb::Object::Types::Table);
+            m_browseTableModel->isEditable();
 
     // Don't allow editing of other objects than tables
```

Comment:
Thank you very much! A simple change to fix this oversight, but it's demonstrating that you've carfully followed the code to pinpoint it. Congratulations!

Reply:
Thank you! It seemed simple and I really wanted that functionality, mainly for the JSON highlighter/editor.